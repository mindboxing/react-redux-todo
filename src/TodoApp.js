import React from 'react';
import AddTodo from './components/AddTodo'
import TodoList from './components/TodoList'
import VisibilityFilters from './components/VisibilityFilters'
import "./styles.css";
import ErrorMessage from './components/ErrorMessage'

export default function TodoApp() {  
  return (
      <div className="todo-app">
          <h1>Todo List</h1>
          <ErrorMessage />
          <AddTodo />
          <TodoList />
          <VisibilityFilters />            
      </div>
  );
}