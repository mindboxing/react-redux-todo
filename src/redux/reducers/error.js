import { ADD_TODO_FAIL } from '../actionTypes';

export const initialState = {
    error: false,
    message: null
}

export const error = (state = initialState, action) => {
  if (!action) {
    return state;
  }
  switch (action.type) {
    case ADD_TODO_FAIL: 
      const resp = action.payload.error.response;
      return (resp) 
      ? { error: true, message: `${resp.status} : ${resp.statusText}`} 
      : state
    default: return {
      error: false,
      message: null
    }
  }
}

export default error;