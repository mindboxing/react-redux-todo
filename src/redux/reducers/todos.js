import { ADD_TODO, ADD_TODO_SUCCESS, TOGGLE_TODO } from '../actionTypes';

export const initialState = {
    allIds: [],
    byIds: {}
}

export const todos = (state = initialState, action) => {
    if (!action) {
      return state;
    }
    switch (action.type) {
        case ADD_TODO_SUCCESS: {
            const { id, content } = action.payload;
            return {
                ...state,
                allIds: [...state.allIds, id],
                byIds: {
                    ...state.byIds,
                    [id]: {
                        content,
                        completed: false
                    }
                }
            }
        };

        case TOGGLE_TODO: {
            const { id } = action.payload;
            return {
                ...state,
                byIds: {
                    ...state.byIds,
                    [id]: {
                        ...state.byIds[id],
                        completed: !state.byIds[id].completed
                    }
                }
            }
        }

        default:
            return state;
    }
}

export default todos;