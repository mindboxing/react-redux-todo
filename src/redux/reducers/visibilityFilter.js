import { SET_FILTER } from "../actionTypes";
import { VISIBILITY_FILTERS } from "../../constants"

export const initialState = VISIBILITY_FILTERS;

const visibilityFilter = (state = initialState, action) => {
  if (!action) {
    return state;
  } 
  switch (action.type) {
    case SET_FILTER: {
      return action.payload.filter;
    }
    default: {
      return state;
    }
  }
}

export default visibilityFilter;