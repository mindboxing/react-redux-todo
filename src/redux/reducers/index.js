import { combineReducers } from 'redux';
import visibilityFilter from './visibilityFilter';
import todos from './todos';
import error from './error';

export default combineReducers({ todos, visibilityFilter, error });