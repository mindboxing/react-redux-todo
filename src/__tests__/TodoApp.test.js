import React from  "react";
import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import TodoApp from '../TodoApp';
import { initialState as todos } from "../redux/reducers/todos";
import { initialState as visibilityFilter } from "../redux/reducers/visibilityFilter";
import { initialState as error } from "../redux/reducers/error";

const mockStore = configureStore([]);
function renderComponent(ui, state={todos, visibilityFilter, error}) {
  const store = mockStore(state);
  return [render((
      <Provider store={store}>
          {ui}
      </Provider>
  )), store];
}

describe('TodoApp', () => {
  test('should render', () => {
    const [element, _] = renderComponent(<TodoApp />);
    expect(element).toMatchSnapshot();
  })  
})
