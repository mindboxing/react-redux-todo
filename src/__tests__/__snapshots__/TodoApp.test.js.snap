// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoApp should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todo-app"
      >
        <h1>
          Todo List
        </h1>
        <div />
        <div>
          <input
            value=""
          />
          <button
            class="add-todo"
          >
            Add Todo
          </button>
        </div>
        <ul
          class="todo-list"
        >
          No todos, yay!
        </ul>
        <div
          class="visibility-filters"
        >
          <span
            class="filter"
          >
            all
          </span>
          <span
            class="filter"
          >
            completed
          </span>
          <span
            class="filter"
          >
            incomplete
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todo-app"
    >
      <h1>
        Todo List
      </h1>
      <div />
      <div>
        <input
          value=""
        />
        <button
          class="add-todo"
        >
          Add Todo
        </button>
      </div>
      <ul
        class="todo-list"
      >
        No todos, yay!
      </ul>
      <div
        class="visibility-filters"
      >
        <span
          class="filter"
        >
          all
        </span>
        <span
          class="filter"
        >
          completed
        </span>
        <span
          class="filter"
        >
          incomplete
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
